### PR TITLE
add generator expressions to matcher for python argumentOrParameter

### LINKED ISSUE #323: Argument scope type doesn't work in python when argument is a generator expression
In the following code:

```python
" ".join(word for word in word_list)
```

Saying "take arg whale" should select the whole generator expression between the parentheses, because it is the argument to `join`. Instead it finds no argument

### PR DESCRIPTION
**Pull request by AndrewDant**
Closes #323

### CODE REVIEW

File: src/languages/python.ts
Comment (Will-Sommers) [inline src/languages/python.ts:85]: Heyo! Cool! 

[Here's the link to the test case recorder](https://www.cursorless.org/docs/contributing/test-case-recorder/). `cursorless record` doesn't recognize well for me, just a heads up. 

A few good tests to record would be: 
- `"chuck arg"` with single argument in list
- `"chuck arg"` with multiple arguments in list, both generator expressions
- `"chuck arg"` with multiple arguments in list, one generator expression one other

I'm not super familiar with python but I assume that there are situations where we can have 2 generator expressions as arguments. If not, disregard that case.

File: src/languages/python.ts
Comment (pokey) [inline src/languages/python.ts:87]: I get the feeling that this one will match any generator expression. So eg it would match the right-hand side of the following:

```python
foo = (bar for baz in bongo)
```

But that's not actually an argument. So I believe you'll need to look at the parent node to make sure you're in an argument list

Comment (Will-Sommers) [reply src/languages/python.ts:87]: Since `generator_expression` is a type of argument, coming from call, would it be `call.arguments[generator_expression]`?

https://github.com/tree-sitter/tree-sitter-python/blob/master/src/node-types.json#L776-L785

Comment (AndrewDant) [reply src/languages/python.ts:87]: Just confirmed this issue.

Comment (AndrewDant) [reply src/languages/python.ts:87]: Would the best path forward be to write my own finder that essentially extends patternFinder() to filter for patterns with the correct parent node type?

Comment (AndrewDant) [reply src/languages/python.ts:87]: Or could existing finders such as ancestorChainNodeFinder() perform this functionality if configured correctly?

Comment (AndrewDant) [reply src/languages/python.ts:87]: Right after asking I started trying it myself and it seems to work better than I expected. I sort of assumed my first thought wouldn't work.

File: src/languages/python.ts
Comment (AndrewDant) [reply src/languages/python.ts:85]: Is the idea behind these kinds of tests to record correct behavior and then running the tests will verify the behavior hasn't changed?

Comment (AndrewDant) [reply src/languages/python.ts:85]: I recorded and pushed a test case for 'take' and 'chuck' for each of the examples you gave. Any others I should add?

File: src/languages/python.ts
Comment (Will-Sommers) [inline src/languages/python.ts:90]: Heyo! I think you could do this using the language matching syntax: 

```
matcher(
      patternFinder("call.generator_expression!"),
      childRangeSelector()
    )
```   
What is happening here is — selecting node if the parent is a `call` type and the child is a `generator_expression` type. Without the `"!"`, we would select all of these nodes, however, we just want to select the arguments, not the entire call.

File: src/languages/python.ts
Comment (Will-Sommers) [reply src/languages/python.ts:85]: Great!

I think a `chuck every argue` would be great. It looks like the `( )` and "," are cleaned up in multi argument situations but it might be good to have a case where we remove every argument when there are two.

I'll defer to @pokey on whether to chuck or keep the `"take"` tests. From my understanding, they're harder to parse since we need to look at the cursor positions to understand the output, and that's tougher than just looking and seeing that the argument has been deleted. If there isn't something in the contrib page, we should add it as a note for contributions.

Comment (pokey) [reply src/languages/python.ts:85]: Our policy is as follows:

We like to have "change" tests for everything. It is not the end of the world to use "take" instead of "change", but we do slightly prefer "change" to "take" for the reasons @Will-Sommers mentions above. Generally not worth re-recording tho imo

For scope types where removal is interesting (eg ensuring comma gets removed), we like to also add "chuck" tests. I think this case does warrant "chuck" tests, as you've done

Comment (pokey) [reply src/languages/python.ts:85]: And agreed a "chuck every" would be nice for good measure

Comment (AndrewDant) [reply src/languages/python.ts:85]: I've added one "chuck every" test with just a generator expression and one with the generator expression and another argument.

File: src/languages/python.ts
Comment (AndrewDant) [reply src/languages/python.ts]: I've made this change and verified it still works.

File: src/languages/python.ts
Comment (AndrewDant) [reply src/languages/python.ts:85]: Also I'll try to remember the preference for change instead of take for tests.

File: src/test/suite/fixtures/recorded/languages/python/chuckEveryArgRed.yml
Comment (pokey) [inline src/test/suite/fixtures/recorded/languages/python/chuckEveryArgRed.yml:1]: This should probably be 
```suggestion
languageId: python
```

File: src/test/suite/fixtures/recorded/languages/python/chuckEveryArgRed.yml
Comment (pokey) [inline src/test/suite/fixtures/recorded/languages/python/chuckEveryArgRed.yml:11]: I guess this one is a good corner case, but would be good to also have a test where there are multiple generator expressions as arguments

File: src/test/suite/fixtures/recorded/languages/python/chuckEveryArgWhale.yml
Comment (pokey) [inline src/test/suite/fixtures/recorded/languages/python/chuckEveryArgWhale.yml:1]:
```suggestion
languageId: python
```

**Review by pokey — COMMENTED** [review]
Almost there!

**Comment by pokey** [comment]
Oh yeah and sorry bout CI; will be fixed once I merge #638

File: src/test/suite/fixtures/recorded/languages/python/chuckEveryArgRed.yml
Comment (AndrewDant) [reply src/test/suite/fixtures/recorded/languages/python/chuckEveryArgRed.yml:1]: Yeah that's definitely not right

**Comment by pokey** [comment]
pre-commit.ci autofix

File: src/test/suite/fixtures/recorded/languages/python/chuckEveryArgRed.yml
Comment (AndrewDant) [reply src/test/suite/fixtures/recorded/languages/python/chuckEveryArgRed.yml:11]: Added to new commit.

File: src/test/suite/fixtures/recorded/languages/python/chuckEveryArgWhale.yml
Comment (AndrewDant) [reply src/test/suite/fixtures/recorded/languages/python/chuckEveryArgWhale.yml]: I re recorded these tests.

File: src/test/suite/fixtures/recorded/languages/python/chuckEveryArgRed.yml
Comment (AndrewDant) [reply src/test/suite/fixtures/recorded/languages/python/chuckEveryArgRed.yml]: I re recorded these tests.

**Comment by AndrewDant** [comment]
I just pushed the latest round of changes.

**Review by pokey — APPROVED** [review]
Good to go!

**Pull request merged**
